src/monmove.cpp

```diff
@@ -344,6 +463,31 @@ void monster::plan()
         }
     }
 
+    //Check if steals, must be idle and have no target in sight.
+    if( ( friendly >= 0 || target != nullptr ) && !fleeing && !has_effect( effect_looting ) ) {
+        //If stealing monster, inventory is empty and with a bit of luck...we search for loot.
+        if( will_steal && inv.empty() && one_in( 100 ) ) {
+            //Find all lootable items within radius
+            std::map<item *, const tripoint> lootable_items = find_loot_in_radius( pos(), MONSTER_LOOT_DIST );
+
+            if( !lootable_items.empty() ) {
+                item_goal = select_desired_loot( lootable_items );
+
+
+                set_dest( item_goal.position() );
+                add_effect( effect_looting, 50_turns );
+            }
+        } //Monster has something in inventory, will steal and with some luck...
+        else if( will_steal && !inv.empty() && one_in( 500 ) ) {
```

Comment:
Probably should consider that most monsters that could be interested in food (wolves?) would just start eating food right after moving close to it, rather than picking it up and eating some time later.

Also I'm not sure how food stealing and eating would work with food in sealed tin cans (or other sealed containers); an animal very likely wouldn't be interested at all in such an item (it doesn't smell like food) and certainly shouldn't be able to eat the food inside unless it can chew through the container.

Reply:
I was going to add the option for animals to eat the item rather than pick it up, but felt may be out of scope. I’ll add this and the checking of containers